packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,290 @@
+} from '../../lib/positioner'
+import TooltipContent from './TooltipContent'
+import { Accessibility } from '../../lib/accessibility/types'
+import { ReactAccessibilityBehavior } from '../../lib/accessibility/reactTypes'
+
+export interface TooltipSlotClassNames {
+  content: string
+}
+
+export interface TooltipState {
+  open: boolean
+}
+
+export interface TooltipProps
+  extends StyledComponentProps<TooltipProps>,
+    ChildrenComponentProps,
+    ContentComponentProps<ShorthandValue>,
+    SimplifiedPositioningProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * */
+  accessibility?: Accessibility
+
+  /** Additional CSS class name(s) to apply.  */
+  className?: string
+
+  /** Existing element the tooltip should be bound to. */
+  mountNode?: HTMLElement
+
+  /** Delay in ms for the mouse leave event, before the tooltip will be closed. */
+  mouseLeaveDelay?: number
+
+  /** A tooltip can show a pointer to trigger. */
+  pointing?: boolean
+
+  /**
+   * Function to render tooltip content.
+   * @param {Function} updatePosition - function to request tooltip position update.
+   */
+  renderContent?: (updatePosition: Function) => ShorthandValue
+
+  /**
+   * DOM element that should be used as tooltip's target - instead of 'trigger' element that is used by default.
+   */
+  target?: HTMLElement
+
+  /** Element to be rendered in-place where the tooltip is defined. */
+  trigger?: JSX.Element
+}
+
+/**
+ * A tooltip that displays information related to an element when the element receives keyboard focus
+ * or the mouse hovers over it.
+ */
+export default class Tooltip extends AutoControlledComponent<TooltipProps, TooltipState> {
+  static displayName = 'Tooltip'
+
+  static className = 'ui-tooltip'
+
+  static slotClassNames: TooltipSlotClassNames = {
+    content: `${Tooltip.className}__content`,
+  }
+
+  static Content = TooltipContent
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      animated: false,
+      as: false,
+      content: false,
+    }),
+    align: PropTypes.oneOf(ALIGNMENTS),
+    defaultOpen: PropTypes.bool,
+    inline: PropTypes.bool,
+    mountNode: customPropTypes.domNode,
+    mouseLeaveDelay: PropTypes.number,
+    open: PropTypes.bool,
+    onOpenChange: PropTypes.func,
+    pointing: PropTypes.bool,
+    position: PropTypes.oneOf(POSITIONS),
+    renderContent: PropTypes.func,
+    target: PropTypes.any,
+    trigger: customPropTypes.every([customPropTypes.disallow(['children']), PropTypes.any]),
+    content: customPropTypes.shorthandAllowingChildren,
+  }
+
+  static defaultProps: TooltipProps = {
+    align: 'center',
+    mountNode: isBrowser() ? document.body : null,
+    position: 'above',
+    mouseLeaveDelay: 500,
+    pointing: true,
+  }
+
+  static autoControlledProps = ['open']
+
+  pointerTargetRef = React.createRef<HTMLElement>()
+  triggerRef = React.createRef<HTMLElement>() as React.MutableRefObject<HTMLElement>
+  tooltipDomElement = null
```

Comment:
I see the usage, but I don't see an assignment. Can you please check?

Reply:
Yep, I removed the ref around the tooltip element. Added, please check.

Reply:
In this case let's use `contentRef`:

```suggestion
  contentRef = React.createRef<HTMLElement>
```